compiler/noirc_evaluator/src/acir/acir_context/mod.rs

```diff
@@ -902,7 +902,7 @@ impl<F: AcirField, B: BlackBoxFunctionSolver<F>> AcirContext<F, B> {
         // `rhs` has the correct bit-size because either it is enforced by the overflow checks
         // or `rhs` is zero when the overflow checks are disabled.
         // Indeed, in that case, rhs is replaced with 'predicate * rhs'
-        self.bound_constraint_with_offset(remainder_var, rhs, predicate, max_rhs_bits, one)?;
+        self.bound_constraint_with_offset(remainder_var, rhs, one, max_rhs_bits, predicate)?;
```

Comment:
Using the predicate as an offset is a small optimisation, because if the predicate is true, then the offset is one and this assert that 'r<rhs', without using a predicate (because 'one' is given for the predicate argument).
If the predicate is false, then this will assert 'r<=rhs', for which 'r=0' is a valid assignment.
So I think we should keep the previous code.

Reply:
I thought that what we are looking for is something like `let r = x % 256;`, where `r` would be between 0 and 255, inclusive, but never 256.

Reply:
This is not true when the predicate is false. For instance `r=x %0` under a false predicate is not between 0 and -1, because this is not possible.